### PR TITLE
[KERNAL] reduce KVARS2 usage by 54 bytes, extend KVECTORS by 14 bytes to make room for 65c816 native vectors

### PR DESCRIPTION
**Pull request by mooinglemur**

### CODE REVIEW

**Review by Fulgen301 — COMMENTED** [review]
This gets rid of machinery that allows the code to work on the 6502. I agree that this is superfluous for the X16 ROM, especially in light of the incoming 65C816 work, but better to remove all of it from the entire KERNAL in one big commit as opposed to removing them piece by piece.

File: kernal/cbm/editor.s [hunk 1]
```diff
@@ -606,7 +593,7 @@ bkln	ldx tblx
 	stx pntr
 	pla
 	pla
-	bne loop2
+	jmp loop2
```
[the remark this answers is not in the document]
Comment (mooinglemur) [reply kernal/cbm/editor.s:596]: The prior code was in effect an unconditional branch.  The high byte of the return address on the stack is always nonzero, so `bne` is working as a `bra`

With the refactor, `loop2` is out of range for a branch.

**Comment by mooinglemur** [comment]
> This gets rid of machinery that allows the code to work on the 6502. I agree that this is superfluous for the X16 ROM, especially in light of the incoming 65C816 work, but better to remove all of it from the entire KERNAL in one big commit as opposed to removing them piece by piece.

I don't think it's worth undoing it here just to do it all at once later.  I don't really favor the all-or-nothing approach as less gets done that way.  I think incremental change is fine, but you're right in that we should work to remove all of it.

File: kernal/cbm/editor.s [hunk 2]
```diff
@@ -219,14 +199,12 @@ cint	jsr iokeys
...
-	cpx nlinesp1    ;done # of lines?
-	bne lps1        ;no...
-	lda #$ff        ;tag end of line table
-	sta ldtb1,x
+; clear screen, populate ldtb1 with non-continuing lines
+clsr	lda #$ff
+	ldx #8
+lps1	sta ldtb1-1,x
+	dex
+	bne lps1
```
[the remark this answers is not in the document]
Comment (mooinglemur) [reply kernal/cbm/editor.s:207]: This is just a style choice.  The existing code works for all nonzero x while yours works with starting x <= 128.

There are plenty of other examples of the original style in the kernal, but I'll accept your suggested change.

File: kernal/cbm/editor.s [hunk 3]
```diff
@@ -1297,3 +1266,12 @@ fkeytb	.byt "LIST:", 13, 0
 
 beeplo: .lobytes 526,885,1404
 beephi: .hibytes 526,885,1404
+
+ldbmap:	
+.repeat 8
+	.byte $01,$02,$04,$08,$10,$20,$40,$80
+.endrepeat
+ldcmap:
```
[the remark this answers is not in the document]
Comment (mooinglemur) [reply kernal/cbm/editor.s:1274]: I agree, I'll go ahead and update those labels to something a little more descriptive.

**Pull request merged**